File: fix_exam.py
import re

with open('src/pages/Examination.tsx', 'r', encoding='utf-8') as f:
    code = f.read()

oldButtonsRegex = re.compile(r"<div style=\{\{ marginBottom: '24px', display: 'flex', gap: '16px', alignItems: 'center' \}\}>[\s\S]*?(?=\{paperData\?\.wordContent)", re.DOTALL)

newButtons = '''<div style={{ marginBottom: '24px', display: 'flex', gap: '16px', alignItems: 'center' }}>
            <button 
              className={tn-} 
              onClick={() => {
                const newData = {...paperData} as any;
                if(newData.blocks) {
                  newData.blocks = newData.blocks.filter(b => b.type !== 'kids_activity');
                }
                delete newData.wordContent;
                setPaperData(newData);
              }}
            >
              Regular Exam Builder (6th-12th)
            </button>
            <button 
              className={tn-} 
              onClick={() => {
                const newData = {...paperData, blocks: [...(paperData?.blocks || []).filter(b => b.type === 'kids_activity'), { id: Math.random().toString(), type: 'kids_activity', category: 'VISUAL_DISCRIMINATION', subType: 'Odd One Out', instruction: 'Circle the odd one out', layoutType: 'grid', items: [], config: { columns: 4 } }]} as any;
                delete newData.wordContent;
                setPaperData(newData);
              }}
            >
              Kids Worksheet Engine (Play-5th)
            </button>
          </div>\n\n          '''

code = oldButtonsRegex.sub(newButtons, code)

oldRenderRegex = re.compile(r"\{paperData\?\.wordContent !== undefined \? \([\s\S]*?\) : paperData\?\.blocks \? \([\s\S]*?\) : \(", re.DOTALL)
newRender = '''{paperData?.blocks && paperData.blocks.some(b => b.type === 'kids_activity') ? (
            <KidsBlockCanvas 
              blocks={paperData.blocks} 
              onChange={(newBlocks) => setPaperData(prev => prev ? {...prev, blocks: newBlocks} : null)} 
            />
          ) : ('''

code = oldRenderRegex.sub(newRender, code)

with open('src/pages/Examination.tsx', 'w', encoding='utf-8') as f:
    f.write(code)

File: fix_exam.js
const fs = require('fs');
let code = fs.readFileSync('src/pages/Examination.tsx', 'utf8');

// Replace the entire button group
const oldButtons =           <div style={{ marginBottom: '24px', display: 'flex', gap: '16px', alignItems: 'center' }}>
            <button 
              className={\tn-\\} 
              onClick={() => {
                const newData = {...paperData} as any;
                delete newData.blocks;
                delete newData.wordContent;
                setPaperData(newData);
              }}
            >
              Use Legacy Form Builder
            </button>
            <button 
              className={\tn-\\} 
              onClick={() => {
                const newData = {...paperData, blocks: paperData?.blocks || []} as any;
                delete newData.wordContent;
                setPaperData(newData);
              }}
            >
              Use Block Canvas (6th-12th)
            </button>
            <button 
              className={\tn-\\} 
              onClick={() => {
                const newData = {...paperData, worksheetElements: paperData?.worksheetElements || []} as any;
                delete newData.blocks;
                setPaperData(newData);
              }}
            >
              Visual Designer (Nursery-5th)
            </button>
          </div>;

const newButtons =           <div style={{ marginBottom: '24px', display: 'flex', gap: '16px', alignItems: 'center' }}>
            <button 
              className={\tn-\\} 
              onClick={() => {
                const newData = {...paperData} as any;
                delete newData.blocks;
                setPaperData(newData);
              }}
            >
              Use Legacy Form Builder
            </button>
            <button 
              className={\tn-\\} 
              onClick={() => {
                const newData = {...paperData, blocks: (paperData?.blocks || []).filter(b => b.type !== 'kids_activity')} as any;
                setPaperData(newData);
              }}
            >
              Use Block Canvas (6th-12th)
            </button>
            <button 
              className={\tn-\\} 
              onClick={() => {
                const newData = {...paperData, blocks: [...(paperData?.blocks || []).filter(b => b.type === 'kids_activity'), { id: Math.random().toString(), type: 'kids_activity', category: 'VISUAL_DISCRIMINATION', subType: 'Odd One Out', instruction: 'Circle the odd one out', layoutType: 'grid', items: [], config: { columns: 4 } }]} as any;
                setPaperData(newData);
              }}
            >
              Kids Worksheet Engine (Play-5th)
            </button>
          </div>;

code = code.replace(oldButtons, newButtons);

// Next, replace the rendering logic
const oldRender =           {paperData?.wordContent !== undefined ? (
            <WordDocumentCanvas 
              content={paperData.wordContent} 
              onChange={(newContent) => setPaperData(prev => prev ? {...prev, wordContent: newContent} : null)} 
            />
          ) : paperData?.blocks ? (
            <BlockCanvas 
              blocks={paperData.blocks} 
              onChange={(newBlocks) => setPaperData(prev => prev ? {...prev, blocks: newBlocks} : null)} 
            />
          ) : (;

const newRender =           {paperData?.blocks && paperData.blocks.some(b => b.type === 'kids_activity') ? (
            <KidsBlockCanvas 
              blocks={paperData.blocks} 
              onChange={(newBlocks) => setPaperData(prev => prev ? {...prev, blocks: newBlocks} : null)} 
            />
          ) : paperData?.blocks ? (
            <BlockCanvas 
              blocks={paperData.blocks} 
              onChange={(newBlocks) => setPaperData(prev => prev ? {...prev, blocks: newBlocks} : null)} 
            />
          ) : (;

code = code.replace(oldRender, newRender);

fs.writeFileSync('src/pages/Examination.tsx', code);
